include/rocksdb/options.h

```diff
@@ -847,6 +847,18 @@ struct DBOptions {
   //
   // Dynamically changeable through SetDBOptions() API.
   bool avoid_flush_during_shutdown = false;
+
+  // If enabled it uses two queues for writes, one for the ones with
+  // disable_memtable and one for the ones that also write to memtable. This
+  // allows the memtable writes not to lag behind other writes. It can be used
+  // to optimize MySQL 2PC in which only the commits, which are serial, write to
+  // memtable.
+  bool separate_queue_for_memtable_writes = false;
+
+  // If true WAL is not flushed automatically after each write. Instead it
+  // relies on manual invocation of FlushWAL to write the WAL buffer to its
+  // file.
+  bool manual_wal_flush = false;
```

Comment:
Can you explain more? The second queue was added after Yi's suggestion, which made the code much cleaner and even improved throughput a bit.
> If we have a direct PrepareToWALBuffer() function, there isn't a second queue, so the extra queue logic can be avoided.